buildchain/buildchain/vagrant.py

```diff
@@ -40,16 +42,53 @@ def mkdir_dot_vagrant() -> None:
         'uptodate': [True],
     }
 
+def task_vagrantupnoprov() -> types.TaskDict:
+    """Run `vagrant up` without provisioning a development environment."""
+    vagrantupnoprov = [config.VAGRANT, 'up']
+    vagrantupnoprov.extend(map(shlex.quote,
+        config.VAGRANT_UP_ARGS + tuple(('--no-provision',))))
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantupnoprov))],
+        'file_dep': [constants.VAGRANT_SSH_KEY_PAIR],
+        'task_dep': [],
+        'uptodate': [False],
+    }
+
+def task_vagrantsnapshot() -> types.TaskDict:
+    """Snapshot development environment."""
+    vagrantsnap = [config.VAGRANT, 'snapshot', 'save', '--force', 'bootstrap', config.VAGRANT_SNAPSHOT_NAME]
```

Comment:
We may want to save the snapshot while the VM is down. Then we'll have no conflict about port assignment.